### PR TITLE
Add custom fields page

### PR DESCRIPTION
**Pull request by JonNorman**
## What does this change?
This adds a commercial admin page that displays all of the [Custom Fields](https://support.google.com/dfp_premium/answer/2694303?hl=en) currently in use in DFP.

I could cache the results in S3 as we do with some other jobs, but I don't think this is high priority / risk enough to warrant adding that in. I have the code ready to do so if anyone disagrees :)

## What is the value of this and can you measure success?
Visibility of what were are doing in DFP - useful for Tom Duffett

## Does this affect other platforms - Amp, Apps, etc?
Nope.

## Screenshots
image
(I didn't really do any of that UX/Styling thing that people talk about)

@guardian/commercial-dev

### CODE REVIEW

**Review by rich-nguyen — APPROVED** [review]
Make sure the page is usable for Tom. Code looks good.

**Comment by JonNorman** [comment]
Yeah, I've tried to locate him - no idea what he wants to do with the page, presume some links might be helpful...

**Comment by JonNorman** [comment]
[Make a simple admin page which lists all the custom fields using DFP api's customfieldservice](https://trello.com/c/4IIxRuVg/36-make-a-simple-admin-page-which-lists-all-the-custom-fields-using-dfp-api-s-customfieldservice)

**Comment by rich-nguyen** [comment]
oh yeah the template! you can deploy this to code, and share it with him, or just show him the page locally.

File: admin/app/dfp/DfpApi.scala [hunk 1]
```diff
@@ -37,6 +37,14 @@ object DfpApi extends Logging {
     })
   }
 
+  def getAllCustomFields: Seq[GuCustomField] = {
+    val stmtBuilder = new StatementBuilder()
+
+    withDfpSession( session => {
+      session.customFields(stmtBuilder).map(toGuCustomField)
+    })
```
Comment (TBonnin) [inline admin/app/dfp/DfpApi.scala]: neatpicking: You can use `{` only
```
withDfpSession { session =>
    session.customFields(stmtBuilder).map(toGuCustomField)
}
```

or even 
`withDfpSession(_.customFields(stmtBuilder).map(toGuCustomField))`

Comment (JonNorman) [inline admin/app/dfp/DfpApi.scala]: Good point - I'll update!

**Comment by JonNorman** [comment]
So there may be a spanner in the works (almost _entirely_ due to me asking Tom what he needed _after_ doing the work for it). Will pick up on Monday.

**Comment by JonNorman** [comment]
Spanner has been removed:

image

**Review by rich-nguyen — APPROVED** [review]

**Pull request merged**